mutagen/ac3.py

```diff
@@ -0,0 +1,308 @@
+# -*- coding: utf-8 -*-
+# Copyright (C) 2019 Philipp Wolfer
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+
+
+"""Pure AC3 file information.
+"""
+
+__all__ = ["AC3", "Open"]
+
+from mutagen._compat import endswith
+from mutagen._file import FileType
+from mutagen._util import (
+    BitReader,
+    BitReaderError,
+    MutagenError,
+    convert_error,
+    loadfile,
+)
+
+
+AC3_HEADER_SIZE = 7
+
+AC3_CHMODE_DUALMONO = 0
+AC3_CHMODE_MONO = 1
+AC3_CHMODE_STEREO = 2
+AC3_CHMODE_3F = 3
+AC3_CHMODE_2F1R = 4
+AC3_CHMODE_3F1R = 5
+AC3_CHMODE_2F2R = 6
+AC3_CHMODE_3F2R = 7
+
+AC3_CHANNELS = {
+    AC3_CHMODE_DUALMONO: 2,
+    AC3_CHMODE_MONO: 1,
+    AC3_CHMODE_STEREO: 2,
+    AC3_CHMODE_3F: 3,
+    AC3_CHMODE_2F1R: 3,
+    AC3_CHMODE_3F1R: 4,
+    AC3_CHMODE_2F2R: 4,
+    AC3_CHMODE_3F2R: 5
+}
+
+AC3_SAMPLE_RATES = [48000, 44100, 32000]
+
+AC3_BITRATES = [
+    32, 40, 48, 56, 64, 80, 96, 112, 128,
+    160, 192, 224, 256, 320, 384, 448, 512, 576, 640
+]
+
+EAC3_FRAME_TYPE_INDEPENDENT = 0
+EAC3_FRAME_TYPE_DEPENDENT = 1
+EAC3_FRAME_TYPE_AC3_CONVERT = 2
+EAC3_FRAME_TYPE_RESERVED = 3
+
+EAC3_BLOCKS = [1, 2, 3, 6]
+
+
+class AC3Error(MutagenError):
+    pass
+
+
+class AC3Info(object):
+
+    """AC3 stream information.
+    The length of the stream is just a guess and might not be correct.
+
+    Attributes:
+        channels (`int`): number of audio channels
+        length (`float`): file length in seconds, as a float
+        sample_rate (`int`): audio sampling rate in Hz
+        bitrate (`int`): audio bitrate, in bits per second
+        type_ (`str`): AC3 or EAC3
+    """
+
+    channels = 0
+    length = 0
+    sample_rate = 0
+    bitrate = 0
+    type_ = 'AC3'
+
+    @convert_error(IOError, AC3Error)
+    @convert_error(IndexError, AC3Error)
+    def __init__(self, fileobj):
+        """Raises AC3Error"""
+        header = bytearray(fileobj.read(6))
+        if not header.startswith(b"\x0b\x77"):
```

Comment:
check the header length here instead of converting IndexError above. convert_error() is mainly for IOError because they can happen in lots of places without a way to handle them directly.